src/main/java/seedu/duke/Quiz.java

```diff
@@ -56,9 +56,13 @@ public static void quizQuestion(Subject subject) throws EscException {
         int max = cards.size();
         int attempted = 0;
         double score = 0;
-        while (attempted < max) {
+        while (attempted < max && attempted != numToQuiz) {
```

Comment:
You check that the number doesn't exceed here but may be better to ensure it before the command stage? You decide how you wanna check for max, just suggestions :))